--- a/arch/arm/mach-sc5xx/spl.c
+++ b/arch/arm/mach-sc5xx/spl.c
@@ -18,6 +18,7 @@
 static int adi_sf_default_speed = CONFIG_SF_DEFAULT_SPEED;
 
 u32 bmode;
+#define BMODE_ADDR 0x82000000
 
 int spl_start_uboot(void)
 {
@@ -48,9 +49,34 @@
 void board_boot_order(u32 *spl_boot_list)
 {
 	const char *bmodestring = sc5xx_get_boot_mode(&bmode);
+	int i = 0;
+	static const char * const bmodes[] = {
+		"JTAG/BOOTROM",
+		"QSPI Master",
+		"QSPI Slave",
+		"UART",
+		"LP0 Slave",
+		"OSPI",
+#ifdef CONFIG_SC59X_64
+		"eMMC",
+#endif
+	};
 
 	printf("ADI Boot Mode: 0x%x (%s)\n", bmode, bmodestring);
 
+	bmode = (*(volatile unsigned int *)BMODE_ADDR) - 1;
+	printf("Entering Developer Mode...\n");
+	if (bmode >= ARRAY_SIZE(bmodes)) {
+		printf("Please enter a bootmode at 0x%08x via JTAG:\n", BMODE_ADDR);
+		for (i = 0; i < ARRAY_SIZE(bmodes); i++)
+			printf("[%d]\t%s\n", i + 1, bmodes[i]);
+
+		/* Allow continuing to load U-Boot proper if needed. */
+		bmode = 0;
+	} else {
+		printf("Booting into %s\n", bmodes[bmode]);
+	}
+
 	switch (bmode) {
 	case 0:
 		printf("SPL execution has completed.  Please load U-Boot Proper via JTAG");
