From 2809f7a41289e6771230f8c96fbab9a784e91ddb Mon Sep 17 00:00:00 2001
From: Nathan Barrett Morrison <nathan.morrison@timesys.com>
Date: Wed, 5 Jun 2019 11:29:50 -0400
Subject: [PATCH] Update initramfs boot sequence for Yocto

---
 include/configs/sc_adi_common.h | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/include/configs/sc_adi_common.h b/include/configs/sc_adi_common.h
index 7d54db004..60662ad4c 100644
--- a/include/configs/sc_adi_common.h
+++ b/include/configs/sc_adi_common.h
@@ -116,6 +116,15 @@
 			__stringify(CONFIG_BAUDRATE) " "\
 	"mem=" CONFIG_LINUX_MEMSIZE
 
+#define CONFIG_BOOTARGS_RAMFS	\
+	"root=/dev/ram " \
+	"clkin_hz=" __stringify(CONFIG_CLKIN_HZ) " " \
+	CONFIG_BOOTARGS_VIDEO \
+	"earlyprintk=serial,uart0,57600 " \
+	"console=ttySC" __stringify(CONFIG_UART_CONSOLE) "," \
+			__stringify(CONFIG_BAUDRATE) " "\
+	"mem=" CONFIG_LINUX_MEMSIZE
+
 #define CONFIG_BOOTARGS_NFS	\
 	"root=/dev/nfs rw " \
 	"nfsroot=${serverip}:${rootpath},tcp,nfsvers=3 " \
@@ -165,15 +174,18 @@
 		"\0" \
 	\
 	"ramfile=zImage\0" \
+	"initramfile=ramdisk.cpio.xz.u-boot\0" \
+	"initramaddr=0xC5000000\0" \
 	"dtbfile=" CONFIG_DTBNAME "\0" \
 	"dtbaddr=" CONFIG_DTBLOADADDR "\0" \
-	"ramargs=set bootargs " CONFIG_BOOTARGS "\0" \
+	"ramargs=set bootargs " CONFIG_BOOTARGS_RAMFS "\0" \
 	"ramboot=" \
 		"tftp ${loadaddr} ${ramfile};" \
 		"tftp ${dtbaddr} ${dtbfile};" \
+		"tftp ${initramaddr} ${initramfile};" \
 		"run ramargs;" \
 		"run addip;" \
-		"bootz ${loadaddr} - ${dtbaddr}" \
+		"bootz ${loadaddr} ${initramaddr} ${dtbaddr}" \
 		"\0" \
 	\
 	"nfsfile=zImage\0" \
-- 
2.11.0

