From 79d5de28fbb6867bd2b0afd756ef9c476f906e4c Mon Sep 17 00:00:00 2001
From: Nathan Barrett Morrison <nathan.morrison@timesys.com>
Date: Mon, 13 May 2019 10:44:01 -0400
Subject: [PATCH] Enable ADI watchdog

---
 include/configs/sc_adi_common.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/sc_adi_common.h b/include/configs/sc_adi_common.h
index da3599c1a..75ef3bf29 100644
--- a/include/configs/sc_adi_common.h
+++ b/include/configs/sc_adi_common.h
@@ -292,11 +292,11 @@
 # endif
 #endif
 
-/*
+
 #define CONFIG_HW_WATCHDOG
 #define CONFIG_ADI_WATCHDOG
 #define CONFIG_WATCHDOG_TIMEOUT_MSECS 10000
-*/
+
 
 #define CONFIG_RSA
 
-- 
2.11.0

