From a61c30e18f498e98ca261198f7f14a5d3c23de13 Mon Sep 17 00:00:00 2001
From: Nathan Barrett-Morrison <nathan.morrison@timesys.com>
Date: Wed, 27 Jan 2021 15:57:32 -0500
Subject: [PATCH] Enable ADI watchdog (SC594)

---
 include/configs/sc_adi_common.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/include/configs/sc_adi_common.h b/include/configs/sc_adi_common.h
index 9e7c136..82bd7c5 100644
--- a/include/configs/sc_adi_common.h
+++ b/include/configs/sc_adi_common.h
@@ -348,11 +348,9 @@
 # endif
 #endif
 
-/*
 #define CONFIG_HW_WATCHDOG
 #define CONFIG_ADI_WATCHDOG
-#define CONFIG_WATCHDOG_TIMEOUT_MSECS 10000
-*/
+#define CONFIG_WATCHDOG_TIMEOUT_MSECS 60000
 
 #define CONFIG_RSA
 
-- 
2.11.0

