From bcdd111ea77440747d86696427f516170c139d84 Mon Sep 17 00:00:00 2001
From: Nathan Barrett Morrison <nathan.morrison@timesys.com>
Date: Wed, 19 Jun 2019 09:01:21 -0400
Subject: [PATCH 2/4] Apply 0002-rename-include-usb_gadgetfs-to-usb-dir.patch

---
 usb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gadgetfs-test/usb.c b/gadgetfs-test/usb.c
index b161154..24a443d 100644
--- a/gadgetfs-test/usb.c
+++ b/gadgetfs-test/usb.c
@@ -34,7 +34,7 @@
 #include <asm/byteorder.h>
 
 #include <linux/types.h>
-#include <linux/usb_gadgetfs.h>
+#include <linux/usb/gadgetfs.h>
 #include <linux/usb/ch9.h>
 
 #ifdef	AIO
-- 
2.11.0
