diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 28bb432..f8508c9 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -8437,10 +8437,12 @@ abi_long do_syscall(void *cpu_env, int num, abi_long arg1,
 #ifdef TARGET_NR_stime /* not on alpha */
     case TARGET_NR_stime:
         {
-            time_t host_time;
-            if (get_user_sal(host_time, arg1))
+           struct timespec ts;
+           ts.tv_nsec = 0;
+           if (get_user_sal(ts.tv_sec, arg1)) {
                 goto efault;
-            ret = get_errno(stime(&host_time));
+           }
+           return get_errno(clock_settime(CLOCK_REALTIME, &ts));
         }
         break;
 #endif
